flow/flow.cpp

```diff
@@ -112,6 +116,104 @@ namespace detail
 
 }
 
+
+
+
+
+// Setup the OpmLog backends
+void setupLogging(int mpi_rank_, const std::string& deck_filename, const std::string& cmdline_output_dir, const std::string& cmdline_output, bool output_cout_, const std::string& stdout_log_id) {
+    // create logFile
+    using boost::filesystem::path;
+    path fpath(deck_filename);
+    std::string baseName;
+    std::ostringstream debugFileStream;
+    std::ostringstream logFileStream;
+
+    // Strip extension "." or ".DATA"
+    std::string extension = boost::to_upper_copy(fpath.extension().string());
+    if (extension == ".DATA" || extension == ".") {
+        baseName = boost::to_upper_copy(fpath.stem().string());
+    } else {
+        baseName = boost::to_upper_copy(fpath.filename().string());
+    }
+
+    std::string output_dir = cmdline_output_dir;
+    if (output_dir.empty()) {
+        output_dir = absolute(path(baseName).parent_path()).string();
+    }
+
+    logFileStream << output_dir << "/" << baseName;
+    debugFileStream << output_dir << "/" << baseName;
+
+    if (mpi_rank_ != 0) {
+        // Added rank to log file for non-zero ranks.
+        // This prevents message loss.
+        debugFileStream << "." << mpi_rank_;
+        // If the following file appears then there is a bug.
+        logFileStream << "." << mpi_rank_;
+    }
+    logFileStream << ".PRT";
+    debugFileStream << ".DBG";
+
+    FileOutputMode output_;
```

Comment:
Some testing with master reveals that this feature is no longer respected. The intention was to be able to run completely output-less, or only with log files, for profiling or other purposes. Running the current master, the eclipse files are always written. Using `--output-mode=none` only turns off the log files, the terminal is still written to. If the last thing is a bug could be discussed, I think "none" should be really nothing, but perhaps we should then also add a "OUTPUT_TERMINAL_ONLY" mode.

The last change would be good to tack onto this PR, which is why I discuss it here. It is in no way this PR's fault that the current master is not behaving properly... Ensuring we turn off eclipse file output when requested to may be a little more involved, or it may be easy, I don't know right now.

Reply:
I will come back to this in a subsequent commit.